v2/devdoc/log_sink_etw_requirements.md

```diff
@@ -20,160 +20,184 @@ The signature of `log_sink_etw.log_sink_log` is:
+  - **SRS_LOG_SINK_ETW_01_049: [** `log_sink_etw.log_sink_log` shall call `log_context_get_property_value_pair_count` to obtain the count of properties that are to be added to the ETW event. **]**
 
-- `_tlgBlobTyp` shall be set to `_TlgBlobEvent4`.
+  - **SRS_LOG_SINK_ETW_01_050: [** `log_sink_etw.log_sink_log` shall call `log_context_get_property_value_pairs` to obtain the properties that are to be added to the ETW event. **]**
 
-- `_tlgChannel` shall be set to 11.
+  - **SRS_LOG_SINK_ETW_01_082: [** If more than 64 properties are given in `log_context`, `log_sink_etw.log_sink_log` shall not add any properties to the event. **]**
 
-- `_tlgLevel` shall be set to the appropriate logging level.
+**SRS_LOG_SINK_ETW_01_042: [** `log_sink_etw.log_sink_log` shall compute the metadata size for the self described event metadata as follows: **]**
 
-- `_tlgOpcode` shall be set to 0.
+- **SRS_LOG_SINK_ETW_01_043: [** Length of the event name (determined at compile time, excluding zero terminator) + 1.
+ **]**
+- **SRS_LOG_SINK_ETW_01_044: [** Length of the `content` field name (determined at compile time, excluding zero terminator) + 1. **]**
 
-- `_tlgKeyword` shall be set to 0.
+- **SRS_LOG_SINK_ETW_01_045: [** Length of the `file` field name (determined at compile time, excluding zero terminator) + 1. **]**
 
-- `_tlgEvtMetaSize` shall be set to the computed metadata size + 4.
+- **SRS_LOG_SINK_ETW_01_046: [** Length of the `func` field name (determined at compile time, excluding zero terminator) + 1. **]**
 
-- `_tlgEvtTag` shall be set to 128.
+- **SRS_LOG_SINK_ETW_01_047: [** Length of the `line` field name (determined at compile time, excluding zero terminator) + 1. **]**
 
-`log_sink_etw.log_sink_log` shall fill the event metadata:
+- **SRS_LOG_SINK_ETW_01_051: [** For each property in `log_context`, the length of the property name + 1 and one extra byte for the type of the field. **]**
 
-- The string `content` (as field name, excluding zero terminator), followed by one byte with the value `TlgInANSISTRING`.
+- **SRS_LOG_SINK_ETW_01_052: [** For struct properties one extra byte shall be added for the field count. **]**
 
-- The string `file` (as field name, excluding zero terminator), followed by one byte with the value `TlgInANSISTRING`.
+**SRS_LOG_SINK_ETW_01_085: [** If the size of the metadata and the formatted message exceeds 4096 bytes, `log_sink_etw.log_sink_log` shall not add any properties to the event. **]**
 
-- The string `func` (as field name, excluding zero terminator), followed by one byte with the value `TlgInANSISTRING`.
+**SRS_LOG_SINK_ETW_01_026: [** `log_sink_etw.log_sink_log` shall fill a `SELF_DESCRIBED_EVENT` structure, setting the following fields: **]**
 
-- The string `line` (as field name, excluding zero terminator), followed by one byte with the value `TlgInINT32`.
+- **SRS_LOG_SINK_ETW_01_027: [** `_tlgBlobTyp` shall be set to `_TlgBlobEvent4`. **]**
```

Comment:
>_tlgBlobTyp [](http://example.com/codeflow?start=34&length=11)

some reference about what this is and means would be nice.

Reply:
Added